### PR TITLE
Split up link and DAD timers; add a SLAAC disabler

### PR DESCRIPTION
**Pull request by hugelgupf**
Split up link and DAD timers; add a SLAAC disabler

### CODE REVIEW

File: pkg/dhclient/dhclient.go [hunk 1]
```diff
@@ -27,6 +28,32 @@ import (
...
+// few seconds later which is different from the address returned by DHCPv6.
+func DisableSLAAC(ifs ...netlink.Link) error {
+	for _, iface := range ifs {
+		name := iface.Attrs().Name
+		autoconfPath := filepath.Join("/proc/sys/net/ipv6/conf", name, "autoconf")
+		if err := ioutil.WriteFile(autoconfPath, []byte("0\n"), 0777); err != nil {
+			return fmt.Errorf("error disabling autoconf on %s: %v", name, err)
+		}
+
+		// Make sure Router Advertisement (RA) remains enabled. The routes are being
```
Comment (rminnich) [inline pkg/dhclient/dhclient.go:47]: I know nothing about how this works but ... even if the autoconf fails, is there a chance you want to try the accept_ra command?

File: pkg/dhclient/dhclient.go [hunk 2]
```diff
@@ -132,6 +162,16 @@ type Config struct {
 	// Retries is how many times to retry DHCP attempts.
 	Retries int
 
+	// LinkUpTimeout is the time to try setting each link up.
+	//
+	// The default is 30 seconds if none is set.
+	LinkUpTimeout time.Duration
```
Comment (rminnich) [inline pkg/dhclient/dhclient.go:168]: just wondering, is there any doc anywhere that says 30 seconds is a good idea?

Comment (rjoleary) [reply pkg/dhclient/dhclient.go:168]: It depends on the network and your network driver. In most cases it should come up quickly, but when it doesn't, drivers generally don't have a great way to report what is going wrong.

File: pkg/dhclient/dhclient.go [hunk 3]
```diff
@@ -27,6 +28,32 @@ import (
...
+		autoconfPath := filepath.Join("/proc/sys/net/ipv6/conf", name, "autoconf")
+		if err := ioutil.WriteFile(autoconfPath, []byte("0\n"), 0777); err != nil {
+			return fmt.Errorf("error disabling autoconf on %s: %v", name, err)
+		}
+
+		// Make sure Router Advertisement (RA) remains enabled. The routes are being
+		// configured by the kernel.
+		acceptRAPath := filepath.Join("/proc/sys/net/ipv6/conf", name, "accept_ra")
+		if err := ioutil.WriteFile(acceptRAPath, []byte("1\n"), 0777); err != nil {
+			return fmt.Errorf("error enabling accept_ra on %s: %v", name, err)
```
Comment (rjoleary) [inline pkg/dhclient/dhclient.go:51]: If one interface is problematic, wouldn't it be more resilient to keep trying the rest?

File: pkg/dhclient/dhclient.go (hunk 1 above)
Comment (rjoleary) [reply pkg/dhclient/dhclient.go:47]: accept_ra should be unconditionally set to 1. It's the only way to get routes in ipv6, and we don't have a userspace implementation in u-root.

File: pkg/dhclient/dhclient.go [hunk 4]
```diff
@@ -48,6 +75,9 @@ func isIpv6LinkReady(l netlink.Link) (bool, error) {
 // IfUp ensures the given network interface is up and returns the link object.
 func IfUp(ifname string, linkUpTimeout time.Duration) (netlink.Link, error) {
 	start := time.Now()
+	if linkUpTimeout == 0 {
+		linkUpTimeout = 30 * time.Second
```
Comment (rjoleary) [inline pkg/dhclient/dhclient.go:79]: Add function-level comment saying the default is 30s.

**Comment by rminnich** [comment]
is there a plan for this? Is it obsolete?

**Comment by rminnich** [comment]
Cleaning out PRs > 1yo

**Pull request closed without merging**